pkg/sql/create_table.go

```diff
@@ -829,9 +829,14 @@ func addIndexForFK(
 	constraintName string,
 	ts FKTableState,
 ) (sqlbase.IndexID, error) {
+	autoIndexName := fmt.Sprintf("%s_auto_index_%s", tbl.Name, constraintName)
```

Comment:
There exist functions (i think `generateUniqueConstraintName`) that make a unique name. We shouldn't error out here if that index doesn't exists, but instead make a new name.